File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamHuongViLinkColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SanPhamHuongViLinkRow } from "./SanPhamHuongViLinkRow";

export interface SanPhamHuongViLinkColumns {
    Id: Column<SanPhamHuongViLinkRow>;
    SanPhamTenSp: Column<SanPhamHuongViLinkRow>;
    HuongViTenHuongVi: Column<SanPhamHuongViLinkRow>;
}

export class SanPhamHuongViLinkColumns extends ColumnsBase<SanPhamHuongViLinkRow> {
    static readonly columnsKey = 'SanPhamDB.SanPhamHuongViLink';
    static readonly Fields = fieldsProxy<SanPhamHuongViLinkColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamHuongViLink/SanPhamHuongViLinkColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.SanPhamHuongViLink")]
[BasedOnRow(typeof(SanPhamHuongViLinkRow), CheckNames = true)]
public class SanPhamHuongViLinkColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string SanPhamTenSp { get; set; }
    public string HuongViTenHuongVi { get; set; }
}